<!-- Basic Tools -->
* Katoh K, Standley DM. 2013. [MAFFT multiple sequence alignment software version 7: improvements in performance and usability](https://dx.doi.org/10.1093/molbev/mst010).
* Price MN, Dehal PS, Arkin AP. 2009. [FastTree 2 â€“ approximately maximum-likelihood trees for large alignments](https://dx.doi.org/10.1371/journal.pone.0009490).
* Yang Z. 2006. [Computational Molecular Evolution](http://download.bioon.com.cn/view/upload/month_0808/20080811_979a0656719f9157b466IruZP7mlp2U9.attach.pdf).
* Kabsch W, Sander C. 1983. [Dictionary of protein secondary structure: pattern recognition of hydrogen-bonded and geometrical features](https://dx.doi.org/10.1002/bip.360221211).
* Muse SV, Gaut BS. 1994. [A likelihood approach for comparing synonymous and nonsynonymous nucleotide substitution rates, with application to the chloroplast genome](http://www.ncbi.nlm.nih.gov/pubmed/7968485).
* Tien MZ, Meyer AG, Sydykova DK, Spielman SJ, Wilke CO. 2013. [Maximum allowed solvent accessibilites of residues in proteins](https://dx.doi.org/10.1371/journal.pone.0080635).
* Ihaka R, Gentleman R. 1996. [R: A language for data analysis and graphics](https://dx.doi.org/10.1080/10618600.1996.10474713).
* Wickham H. 2009. [ggplot2: elegant graphics for data analysis](http://had.co.nz/ggplot2/book).
* Berman HM, Westbrook J, Feng Z, Gilliland G, Bhat TN, Wessig H, Shindyalov IN, Bourne PE. 2000. [The Protein Data Bank](https://dx.doi.org/10.1093/nar/28.1.235). Nucleic Acids Res. 28:88-95.
* Berman HM. 2008. [The Protein Data Bank: a historical perspective](https://dx.doi.org/10.1107/S0108767307035623). Acta Crystallographica Section A: Foundations of Crystallography A64.
* Kabsch W, Sander C. 1983. [Dictionary of protein secondary structure: pattern recognition of hydrogen-bonded and geometrical features](http://www.ncbi.nlm.nih.gov/pubmed/6667333). Biopolymers 22:2577-2637.
* Kosakovsky-Pond SL, Frost SDW, Muse SV. 2005. [HyPhy: hypothesis testing using phylogenies](https://dx.doi.org/10.1093/bioinformatics/bti079). Bioinformatics 21:676-679.
* Cock PJ, Antao T, Chang JT, Chapman BA, Cox CJ, Dalke A, Friedberg I, Hamelryck T, Kauff F, Wilczynski B, de Hoon MJ. 2009. [Biopython: freely available Python tools for computational molecular biology and bioinformatics](https://dx.doi.org/10.1093/bioinformatics/btp163). Bioinformatics 25:1422-1423.

<!-- Structural Constraints -->
* Scherrer MP, Meyer AG, Wilke CO. 2012. [Modeling coding-sequence evolution within the context of residue solvent accessibility](https://dx.doi.org/10.1186/1471-2148-12-179). BMC Evol. Biol. 12:179.
* Meyer AG, Wilke CO. 2012. [Integrating sequence variation and protein structure to identify sites under selection](http://dx.doi.org/10.1093/molbev/mss217). Mol. Biol. Evol. 30:36-44.
* Meyer AG, Dawson ET, Wilke CO. 2013. [Cross-species comparison of site-specific evolutionary-rate variation in influenza hemagglutinin](http://dx.doi.org/10.1098/rstb.2012.0334). Phil. Trans. R. Soc. B 368:20120334.
* Ramsey DC, Scherrer MP, Wilke CO. 2011. [The relationship between relative solvent accessibility and evolutionary rate in protein evolution](https://dx.doi.org/10.1534/genetics.111.128025). Genetics 188:479-488.
* Bustamante CD, Townsend JP, Hartl DL. 2000. [Solvent Accessibility and Purifying Selection Within Proteins of Escherichia coli and Salmonella enterica](http://www.ncbi.nlm.nih.gov/pubmed/10677853). Mol. Biol. Evol. 17:301-308.
* Franzosa EA, Xia Y. 2009. [Structural Determinants of Protein Evolution Are Context-Sensitive at the Residue Level](https://dx.doi.org/10.1093/molbev/msp146). Mol. Biol. Evol. 26:2387-2395
* Grahnen JA, Nandakumar P, Kubelka J, Liberles DA. 2011. [Biophysical and structural consideration for protein sequence evolution](https://dx.doi.org/10.1186/1471-2148-11-361). BMC Evol. Biol. 11:2455-2464.
* Sikosek T, Chan HS. 2014. [Biophysics of protein evolution and evolutionary protein biophysics](https://dx.doi.org/10.1098/rsif.2014.0419). J. Royal Soc. Interface 11:20140419.
* Shahmoradi A, Sydykova DK, Spielman SJ, Jackson EL, Dawson ET, Meyer AG, Wilke CO. 2014. [Predicting evolutionary site variability from structure in viral proteins: buriedness, packing, flexibility, and design](https://dx.doi.org/10.1007/s00239-014-9644-x). J. Mol. Evol. 79:130-142.
* Echave J, Jackson EL, Wilke CO. 2014. [Relationship between protein thermodynamic constraints and variation of evolutionary rates among sites](https://dx.doi.org/10.1088/1478-3975/12/2/025002). Phys. Biol. 12:025002. 
* Huang TT, Marcos ML, Hwang JK, Echave J. 2014. [A mechanistic stress model of protein evolution accounts for site-specific evolutionary rates and their relationship with packing density and flexibility](https://dx.doi.org/10.1186/1471-2148-14-78). BMC Evol. Biol. 78:14.
