* Katoh K, Standley DM. 2013. [MAFFT multiple sequence alignment software version 7: improvements in performance and usability](https://dx.doi.org/10.1093/molbev/mst010).
* Price MN, Dehal PS, Arkin AP. 2009. [FastTree 2 â€“ approximately maximum-likelihood trees for large alignments](https://dx.doi.org/10.1371/journal.pone.0009490).
* Yang Z. 2006. [Computational Molecular Evolution](http://download.bioon.com.cn/view/upload/month_0808/20080811_979a0656719f9157b466IruZP7mlp2U9.attach.pdf).
* Kabsch W, Sander C. 1983. [Dictionary of protein secondary structure: pattern recognition of hydrogen-bonded and geometrical features](https://dx.doi.org/10.1002/bip.360221211).
* Muse SV, Gaut BS. 1994. [A likelihood approach for comparing synonymous and nonsynonymous nucleotide substitution rates, with application to the chloroplast genome](http://www.ncbi.nlm.nih.gov/pubmed/7968485).
* Tien MZ, Meyer AG, Sydykova DK, Spielman SJ, Wilke CO. 2013. [Maximum allowed solvent accessibilites of residues in proteins](https://dx.doi.org/10.1371/journal.pone.0080635).
* Ihaka R, Gentleman R. 1996. [R: A language for data analysis and graphics](https://dx.doi.org/10.1080/10618600.1996.10474713).
* Wickham H. 2009. [ggplot2: elegant graphics for data analysis](http://had.co.nz/ggplot2/book).

